src/config/watcher.rs

```diff
@@ -127,7 +127,11 @@ fn create_watcher(
 #[cfg(unix)]
 fn add_paths(watcher: &mut RecommendedWatcher, config_paths: &[PathBuf]) -> Result<(), Error> {
     for path in config_paths {
-        watcher.watch(path, RecursiveMode::NonRecursive)?;
+        if path.is_dir() {
+            watcher.watch(path, RecursiveMode::Recursive)?;
+        } else {
+            watcher.watch(path, RecursiveMode::NonRecursive)?;
+        }
```

Comment:
```suggestion
        watcher.watch(path, RecursiveMode::Recursive)?;
```

Looking at https://docs.rs/notify/latest/notify/trait.Watcher.html#tymethod.watch it looks like the `RecursiveMode` parameter is only used if the path is a directory and ignored if it is a file so I think we can just always set it.